src/core.cc

```diff
@@ -660,8 +660,12 @@ struct text_object *construct_text_object(char *s, const char *arg, long line,
   END OBJ(obsd_vendor, 0) obj->callbacks.print = &get_obsd_vendor;
   END OBJ(obsd_product, 0) obj->callbacks.print = &get_obsd_product;
 #endif /* __OpenBSD__ */
-  END OBJ(buffers, &update_meminfo) obj->callbacks.print = &print_buffers;
-  END OBJ(cached, &update_meminfo) obj->callbacks.print = &print_cached;
+  END OBJ(buffers, &update_meminfo)  obj->data.s = strndup(arg ? arg : "", text_buffer_size.get(*state));
```

Comment:
This also seems like it could be generalized. Either replace with a function, or a macro if you must.